lib/phoenix_html/link.ex

```diff
@@ -165,4 +175,32 @@ defmodule Phoenix.HTML.Link do
 
     {value, opts}
   end
+
+  defp valid_destination!(to, calling_func) do
+    if invalid_destination?(to) do
+      raise ArgumentError, """
+      unsupported scheme given to #{calling_func}. In case you want to link to an
+      unknown or unsafe scheme, such as javascript, use a tuple: {:javascript, rest}
+      """
+    end
+
+    case to do
+      {_, dest} -> dest
```

Comment:
This will unfortunately unwrap a `safe` tuple causing the input to possibly be escaped twice (or be escaped when it should not).